Sources/Apollo/ApolloClient.swift

```diff
@@ -123,6 +123,26 @@ public class ApolloClient {
     return send(operation: mutation, context: context, handlerQueue: queue, resultHandler: resultHandler)
   }
 
+    /// Subscribe to a subscription
+    ///
+    /// - Parameters:
+    ///   - subscription: The subscription to subscribe to.
+    ///   - queue: A dispatch queue on which the result handler will be called. Defaults to the main queue.
+    ///   - resultHandler: An optional closure that is called when mutation results are available or when an error occurs.
+    /// - Returns: An object that can be used to cancel an in progress mutation.
+    public func subscribe<Query: GraphQLSubscription>(subscription: Query, queue: DispatchQueue = DispatchQueue.main, resultHandler: @escaping OperationResultHandler<Query>) -> GraphQLSubscriptionWatcher<Query> {
```

Comment:
`Query` type parameter should be `Subscription` I think.

Reply:
Agree

Reply:
Yeah, `HTTPNetworkTransport` should probably check for supported operation types and throw an error for subscriptions.